rails/es6.md

```diff
@@ -0,0 +1,105 @@
+## [Home](../README.md)
+
+## Summary
+
+Including ES6 in Rails applications can be very useful. Using JavaScript
+packages that are only available/maintained in npm version has traditionally
+been a problem with Rails application. This documents an approach chosen among a
+couple of different options.
+
+We settled upon sprockets-commoner over webpack and browserify because it
+integrates with the Rails asset pipeline and least modifies the natural
+development of JavaScript components.
+
+### Steps to upfit your Rails application
+  * add `'expose window.[component_name]'` to the top of ES6 class files
+  * add `gem 'sprockets-commoner'` to Gemfile
+  * add `.babelrc` to project root
+    * add other presets as necessary (eg. react)
+
+  ```json
+    {
+      presets: [
+        "es2015"
+      ]
+    }
+  ```
+  * add NPM dependencies to `package.json` (including react-dom, react-chartjs, etc.)
+
+  ```json
+    "dependencies": {
+      "autoprefixer": "^6.3.5",
+      "axios": "^0.9.1",
+      "babel-core": "^6.14.0",
+      "babel-generator": "^6.14.0",
+      "babel-helpers": "^6.8.0",
+      "babel-polyfill": "^6.7.4",
+      "babel-types": "^6.14.0",
+      "babel-preset-es2015": "^6.9.0",
+      "babel-preset-stage-0": "^6.5.0",
+      "babel-runtime": "^6.6.1",
+      "jquery": "^2.2.2",
+      "jquery-ujs": "^1.2.1",
+      "lodash": "^4.7.0"
+    }
+  ```
+
+  * add `app/assets/config/manifest.js` file
+
+  ```js
+  //= link_tree ../images
+  //= link_directory ../javascripts .js
+  //= link_directory ../stylesheets .css
+  ```
+
+#### React specific steps
+  * rename .jsx files to .js
+  * move propTypes out of class, and change syntax to `[class_name].propTypes = { ... }`
```

Comment:
I don't remember what happened other than it didn't work. So, I'm not sure what it was related to. You are welcome to give it a try and see. You'll find it in an unmerged branch on the project we tried it.